src/metabase/query_processor/streaming/csv.clj

```diff
@@ -29,10 +29,13 @@
   (let [writer             (BufferedWriter. (OutputStreamWriter. os StandardCharsets/UTF_8))
         ordered-formatters (volatile! nil)]
     (reify qp.si/StreamingResultsWriter
-      (begin! [_ {{:keys [ordered-cols results_timezone]} :data} viz-settings]
+      (begin! [_ {{:keys [ordered-cols results_timezone format-rows?]
+                   :or   {format-rows? true}} :data} viz-settings]
         (let [col-names (common/column-titles ordered-cols (::mb.viz/column-settings viz-settings))]
           (vreset! ordered-formatters (mapv (fn [col]
-                                              (formatter/create-formatter results_timezone col viz-settings))
+                                              (if format-rows?
+                                                (formatter/create-formatter results_timezone col viz-settings)
+                                                identity))
```

Comment:
Let's check this condition once outside the `mapv`. Like:
```clojure
(cond->> ordered-cols
  format-rows? (mapv #(formatter/create-formatter results_timezone % viz-settings)))
```

Reply:
Good point.
Since this is creating formatting functions, I can't quite do it the way you've suggested, but I've captured the spirit of the idea.